### PR TITLE
add snap option

### PR DESCRIPTION
**Pull request by kerautret**
# PR Description

Add SnapShot option for meshViewer and 3dVolViewer (useful to get visualisation without interaction like for scripting and/or online demonstration)

# Checklist

- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Check if it follows the tools structure described in [CONTRIBUTING.md](https://github.com/DGtal-team/DGtalTools-contrib/blob/master/CONTRIBUTING.md)
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtalTools-contrib/blob/master/ChangeLog.md) added.
- [x] Update the readme with potentially a screenshot of the tools if it applies. 
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).

### CODE REVIEW

**Comment by kerautret** [comment]
@akrah could you please test this -d option in graphViewer to be sure that works on linux ? (to check the script of centreline also in linux)

**Comment by akrah** [comment]
I tested and it doesn't work: the value of "viewer.snapshotCounter()" line 232 is 0.

File: visualisation/graphViewer.cpp [hunk 1]
```diff
@@ -58,7 +79,8 @@ int main( int argc, char** argv )
     ("addMesh,m", po::value<std::string>(), "add mesh in the display.")
     ("meshColor", po::value<std::vector<unsigned int> >()->multitoken(), "specify the color mesh.")
     ("edgeColor", po::value<std::vector<unsigned int> >()->multitoken(), "specify the color of edges.")
-    ("colormap,c", "display vertex colored by order in vertex file or by radius scale if the radius file is specidfied (-r).");
+    ("colormap,c", "display vertex colored by order in vertex file or by radius scale if the radius file is specidfied (-r).")
+    ("doSnapShotAndExit,d", po::value<std::string>(), "save display snapshot into file. Notes that the camera setting is set by default according the last saved configuration (use SHIFT+Key_M to save current camera setting in the Viewer3D). If the camera setting was not saved it will use the default camera setting." );
```
Comment (akrah) [inline visualisation/graphViewer.cpp]: It's "SHIFT+L" on my computer: it could depend on each user ?

**Comment by kerautret** [comment]
and have you tested the actual meshViewer on the master did it works with -d ? you have QT 4?

File: visualisation/graphViewer.cpp (hunk 1 above)
Comment (akrah) [inline visualisation/graphViewer.cpp]: Linux, of course ^^

**Comment by akrah** [comment]
Do you speak about MeshViewerEdit ?

**Comment by kerautret** [comment]
no meshViewer

**Comment by akrah** [comment]
Not in this repo but in the DGtalTools repo ?

**Comment by kerautret** [comment]
yes DGtalTools ;) (committed and merged fews days ago)

**Comment by akrah** [comment]
Same problem.
After some tests, it appears that the slot "saveSnapshot()" is never called... but it could be called after the "application.exec()" instruction.

**Comment by kerautret** [comment]
strange thanks, yes after application.exec() but it is too late in this case.( for scripting)
But its works on debian with last qglversion so perhaps a qglviewe version version pb, I merge for now perhaps putting an issue if we reproduce the pb on other distribution...

**Pull request merged**